hw4/02-sessions.js

```diff
@@ -1,17 +1,51 @@
 const express = require('express');
 const session = require('express-session');
 const app = express();
-const port = process.env.PORT || 5000;
+const port = process.env.PORT || 5001;
+// Use the express-session module
+// var sess;
+app.use(
+    session({
+        store: new session.MemoryStore(),
+        secret: 'keyboard cat',
+        resave: false,
+        saveUninitialized: true,
+        cookie: { secure: false },
+    })
+);
 
-// Add your code here
+// app.get('/', (req, res, next) => {
+//     // Add your code here
+//     //  sess.route = '';
+//     res.send('Hello');
+// });
+app.get('/favicon.ico', (req, res) => res.status(204));
 
-// Use the express-session module
-app.use(/** ... */);
+//Middleware gets executed for every route
+app.use((req, res, next) => {
+    res.set({ 'Content-Type': 'text/plain' });
+    res.write(`Currently on route ${req.url}\n\n`);
+    if (req.session.previouslyVisited) {
+        req.session.previouslyVisited.push(`${req.url}`);
```

Comment:
Here when a user visits a route it's also showing up in the "Previously Visited" section because you're pushing it to the previouslyVisited array before writing it to the response. Based on the screenshots you should not push it to the previouslyVisited array until after you write it

Reply:
@vuong2-pdx will update it. Please check it again, I am new to this.